pkg/analytics/client_test.go

```diff
@@ -56,14 +62,156 @@ func TestAnalytics_Hash(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			assert := assert.New(t)
 			analytics := &Client{
-				UserId: userId,
+				userId: userId,
 			}
 			actual := analytics.Hash(tt.given)
 			assert.Equal(tt.expect, actual)
 		})
 	}
 }
 
-type jsonConvertable struct {
-	Foo string `json:"foo"`
+func TestAnalyticsSend(t *testing.T) {
+	cases := []struct {
+		name   string
+		send   func(client *Client)
+		expect []sentPayload
+	}{
+		{
+			name: "direct send at level info with properties",
+			send: func(c *Client) {
+				c.userId = "my-user@klo.dev"
+				c.AppendProperties(map[string]any{"property_1": "aaa"})
+				c.Info("hello world")
+			},
+			expect: []sentPayload{{
+				"id":    "my-user@klo.dev",
+				"event": "hello world",
+				"properties": map[string]any{
+					"_logLevel":  "info",
+					"status":     "info",
+					"validated":  false,
+					"property_1": "aaa",
+				},
+			}},
+		},
+		{
+			name: "send via logger with no fields",
+			send: func(c *Client) {
+				c.userId = "my-user@klo.dev"
+				logger := zap.New(c.NewFieldListener(zapcore.WarnLevel))
```

Comment:
nit: this is more of a test for the FieldListener

Reply:
yeah, good point. This is one of those "my brain wasn't quite working right" things. :) Will fix.

Reply:
The part I was actually interested in was the non-global properties, so I just wrote that. FieldListener is as untested as it was before.